packages/components/organisms/f-checkout/src/services/tests/mapper.test.js

```diff
@@ -0,0 +1,95 @@
+import mapCheckoutRequest from '../mapper';
+
+const defaultParams = {
+    address: {},
+    customer: {},
+    isCheckoutMethodDelivery: true,
+    time: {},
+    userNote: ''
+};
+
+describe('checkout mapper', () => {
+    it('should map customer details correctly', () => {
+        // Arrange
+        const customer = {
+            firstName: 'Test',
+            lastName: 'Tester',
+            mobileNumber: '07890123456'
+        };
+
+        // Act
+        const requestBody = mapCheckoutRequest({
+            ...defaultParams,
+            customer
+        });
+
+        // Assert
+        expect(requestBody.customer.firstName).toBe(customer.firstName);
```

Comment:
You should be able to pull out the values you need from above and just reference them here? e.g the `firstName, lastName etc`

Reply:
Do you mean just like this?
```suggestion
        expect(requestBody.customer.firstName).toBe('Test');
```

Reply:
I mean from the requestBody call? Would just help in these expects as you can just reference the const value instead of navigating the three levels down. Not a biggy tho.

Reply:
Ah I see what you mean, something like this?:

```javascript
const { customer } = mapCheckoutRequest({
    ...defaultParams,
    customer
});

const {
    firstName, lastName, phoneNumber, dateOfBirth
} = customer;

expect(firstName).toBe(customer.firstName);
expect(lastName).toBe(customer.lastName);
expect(phoneNumber).toBe(customer.mobileNumber);
expect(dateOfBirth).toBe(null);
```

Reply:
The structure, if we care about it, which I think shouldn't matter, could be tested. In this test we only really care about the customer details being mapped properly & not the actual structure of the response. The `customer` object can be described as the `thing` that holds the key bits of information we need so i think that should be enough to cover what the nested response outputs.